suzieq/engines/pandas/engineobj.py

```diff
@@ -156,7 +158,10 @@ def get_valid_df(self, table: str, **kwargs) -> pd.DataFrame:
         # We cannot use sets because we want to preserve the order of fields
         filter_fields = [x for x in self.iobj.get_filter_fields
                          if x not in addnl_fields]
-        fields += [x for x in filter_fields+addnl_fields if x not in fields]
+        user_query_cols = [x for x in self._get_user_query_cols(query_str)
```

Comment:
Why we inject the query string fields both in `get()` and `get_valid_df()` wouldn't doing it only in the latter be enough?

Reply:
Good point, fixing.